src/plugins/gs-plugin-eos.c

```diff
@@ -83,6 +88,57 @@ on_desktop_apps_changed (GDBusConnection *connection,
 	g_hash_table_destroy (apps);
 }
 
+static void
+reload_default_branches (GsPlugin *plugin)
+{
+	GsPluginData *priv = gs_plugin_get_data (plugin);
+	g_autoptr(GKeyFile) config_file = g_key_file_new ();
+	g_autoptr(GError) error = NULL;
+	g_auto(GStrv) groups = NULL;
+	GStrv group;
```

Comment:
Can you just do the iteration with an index instead? And check for `groups[index] == NULL` as the termination condition